index.js

```diff
@@ -1,117 +1,142 @@
 const express = require("express");
+const morgan = require("morgan");
+const bodyParser = require("body-parser");
+const uuid = require("uuid");
+
 const app = express();
-const morgan = require("morgan")
-bodyParser = require('body-parser'),
-      uuid = require('uuid');
-
-    const app = express();
-    app.use(bodyParser.json());
-
-    let topMovies = [
-{
- id: 1,
- title: 'harry potter and the sorcerer\'s stone',
- author: 'J.K. Rowling',
- director: 'this person',
- genre: 'this genre'
-},
-{
- id: 2,
- title: 'Lord of the rings',
- author:'J.R.R. Tolkien',
- director: 'this person',
- genre: 'this genre'
-},
-{
- id: 3,
- title: 'Twilight',
- author: 'stephanie Meyer',
- director: 'this person',
- genre: 'this genre'
- }
-];
+app.use(morgan("common"));
+app.use(bodyParser.json());
 
-app.use(morgan("common")); 
 app.use(express.static("public"));
 
-// gets the data about a single director, by name \\
+let topMovies = [
+  {
+    id: 1,
+    title: "harry potter and the sorcerer's stone",
+    author: "J.K. Rowling",
+    director: "this person",
+    genre: "this genre",
+  },
+  {
+    id: 2,
+    title: "Lord of the rings",
+    author: "J.R.R. Tolkien",
+    director: "this person",
+    genre: "this genre",
+  },
+  {
+    id: 3,
+    title: "Twilight",
+    author: "stephanie Meyer",
+    director: "this person",
+    genre: "this genre",
+  },
+];
 
-app.get('/movies/:director', (req, res) => {
-  res.json(topMovies.find((movies) => {
-  return movies.director ===req.params.director
```

Comment:
Given that one of the requirements of the exercise is to implement an endpoint to fetch a single movie by its title. I updated the route  `/movies/:director` to `/movies/director/:director` and added a new route for `/movies/:title`. 

From Express perfective `/movies/:title` and `/movies/:director` are technically the same route and that would've caused a bug because for Express both routes have the same format `/movies/:some_keyword_chosen_by_the_dev`. The routes must be unique and the part of the URL the name used for the arguments which is the part that has a colon is not taken into account when Express determines if a route is unique. 

As an example, all these routes are equivalent and could cause unexpected bugs more than one of them are defined in the same script: `/movies/:genre`, `/movies/:date`,`/movies/:id`,`/movies/:anything`. Unless the use different actions such as` app.get("/movies/:title"` and `app.delete("/movies/:id"`. Althought these two routes are technically the sane because the same structure (`/movies/:name_of the_argument`), they are considered to be different by express because one uses the `GET` method and the other one uses the `DELETE` method